libs/execution_context/ebpf_program.c

```diff
@@ -1614,14 +1614,26 @@ _ebpf_program_verify_program_info_hash(_In_ const ebpf_program_t* program)
         goto Exit;
     }
 
-    if (output_hash_length != program->parameters.program_info_hash_length) {
-        result = EBPF_INVALID_ARGUMENT;
-        goto Exit;
-    }
+    if (program->parameters.program_info_hash_length == 0) {
+        // This is the first time the program info hash is being computed.
```

Comment:
Why is this done on first attach rather than at verification time?

Reply:
Add comments explaining why this is the correct behavior. 

An instance of an ebpf_program_t is bound to a specific program information provider, which is fixed when the object is initially created by the user process. This locks the program information provider to that specific program information provider version. The verifier then queries the program object to get the program information, which is essentially immutable (if it changes it won't rebind).